### ISSUE TITLE
`DataFrame.set_index` stub return hint inconsistent with method definition

### ISSUE THREAD

**Issue by bzoracler** [VirtusLab/pandas-stubs#20]
Return hint of the stub is `DataFrame`:

https://github.com/VirtusLab/pandas-stubs/blob/425b0bf4f3e31115aaa8c384024d3b928628bef2/third_party/3/pandas/core/frame.pyi#L134

Based on the return value of Pandas `1.0.x` API, the return hint should be `Optional[DataFrame]`:

https://github.com/pandas-dev/pandas/blob/1.0.x/pandas/core/frame.py#L4366-L4367

Note that this method's docstring in the `1.0.x` API was incorrect (https://github.com/pandas-dev/pandas/blob/1.0.x/pandas/core/frame.py#L4210-L4213), this was fixed in the commit https://github.com/pandas-dev/pandas/commit/b110046099014a63217042478393e715a083604b# , resulting in

https://github.com/pandas-dev/pandas/blob/b110046099014a63217042478393e715a083604b/pandas/core/frame.py#L4536-L4539

```
        Returns
        -------
        DataFrame or None
            Changed row labels or None if ``inplace=True``.
```

**Comment by zkrolikowski-vl** [comment]
Thanks.

I'll add some tests snippets for this and some other methods from [pandas-dev/pandas@b110046](https://github.com/pandas-dev/pandas/commit/b110046099014a63217042478393e715a083604b), and release a new version.

If you run into any more problems you're more than welcome to propose any changes.

**Comment by zkrolikowski-vl** [comment]
This issue is fixed in version 1.0.4.5.
